File: android/src/main/kotlin/vn/thanhpt/simple_paypal/EnvironmentPaypal.kt
package vn.thanhpt.simple_paypal

import com.paypal.android.corepayments.Environment

enum class EnvironmentPaypal(val value: String) {
    SANDBOX("SANDBOX"),
    LIVE("LIVE");

    companion object {
        private fun fromString(name: String): EnvironmentPaypal {
            return values().find { it.value.equals(name, ignoreCase = true) } ?: SANDBOX
        }

        fun toPaypalEnvironment(name: String): Environment {

            val env = fromString(name)

            return when (env) {
                SANDBOX -> Environment.SANDBOX
                LIVE -> Environment.LIVE
            }
        }
    }
}

File: lib/environment_paypal.dart
enum EnvironmentPaypal { sandbox, live }

extension EnvironmentPaypalExtension on EnvironmentPaypal {
  String get value {
    switch (this) {
      case EnvironmentPaypal.sandbox:
        return 'SANDBOX';
      case EnvironmentPaypal.live:
        return 'LIVE';
    }
  }
}
